modules/tests/src/main/scala/ch/epfl/scala/debugadapter/testfmk/DebugTestSuite.scala

```diff
@@ -83,15 +85,15 @@ trait DebugTest {
     }
   }
 
-  private def check(client: TestingDebugClient, attach: Option[Int])(steps: DebugStepAssert[?]*): Unit = {
+  private def check(client: TestingDebugClient, attach: Option[Int])(steps: DebugStepAssert*): Unit = {
     client.initialize()
     attach match {
       case Some(port) => client.attach("localhost", port)
       case None => client.launch()
     }
 
     val sourceBreakpoints = steps
-      .map(_.step)
+      .collect { case s: SingleStepAssert[_] => s.step }
```

Comment:
In Scala 3, we prefer the new `[?]` syntax than `[_]`

```suggestion
      .collect { case s: SingleStepAssert[?] => s.step }
```